enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,363 @@
+}
+
+// ManagedClusterScalarSpec defines the attributes of the ManagedClusterScalar.
+type ManagedClusterScalarSpec struct {
+	// PrioritizerName will be the prioritizer name used in placement.
+	// Each ManagedClusterScalar only represent the scalar value of one prioritizer for one cluster.
+	// +kubebuilder:validation:Required
+	// +required
+	PrioritizerName string `json:"prioritizerName,omitempty"`
+}
+
+//ManagedClusterScalarStatus represents the current status of ManagedClusterScalar.
+type ManagedClusterScalarStatus struct {
+	// Conditions contain the different condition statuses for this managed cluster scalar.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Scalar contains a scalable value of this managed cluster.
+	// +optional
+	Scalar int64 `json:"scalar,omitempty"`
+
+	// ValidUntil defines the time this scalar is valid.
+	// After this time, the scalar is considered to be invalid by placement. nil means never expire.
+	// The controller ownning this resource should keep the scalar up-to-date.
+	// +optional
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	ValidUntil *metav1.Time `json:"validUntil" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedClusterScalarList is a collection of managed cluster scalar.
+type ManagedClusterScalarList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is a list of managed clusters
+	Items []ManagedClusterScalar `json:"items"`
+}
+```
+
+### Let placement prioritizer maintain the lifecycle of the CR.
+- Create
+
+The creation of ManagedClusterScalar CR is handled by each prioritizer plugin on demand. As every update in ManagedClusterScalar CR will introduce an api call, that will bring potential performance presure to api server. So we only want to create ManagedClusterScalar CRs for enabled prioritizer in filterd cluster namespace.
+
+The placement schedule framework will filter avaliable clusters as first step.
+Then the framework will call each prioritizer one by one. If the prioritizer is enable in placement yaml, it will create a corresponding ManagedClusterScalar CR in filtered cluster namespace as below. 
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterScalar
+metadata:
+  name: {PrioritizerName}
+  namespace: cluster1
+spec:
+  prioritizerName: {PrioritizerNameInPlacement}
+```
+The name is the prioritizer name.
+The namespace is the filtered cluster namespace.
+The spec.prioritizerName is the prioritizer name used in placement yaml. 
+
+The code change includes add PreScore() in Prioritizer interface, in which to create ManagedClusterScalar CR before Score().
```

Comment:
If you treated the presence of the addon prioritizer plugin in the managedcluster as sufficient to write data back, then PreScore is unnecessary.  Also, Pre versus Post versus During isn't really material since the status is reported back asynchronously, right?

Reply:
Hmm, yes. Let me update this part, we may only want PreScore to do some check before Score, not to handle the ManagedClusterScalar CR creation.